### PR TITLE
network: Add checks for advanced network options

### PR DESCRIPTION
**Pull request by RytoEX**
### Description

Add checks for Network Optimizations and TCP Pacing.

### Motivation and Context

Detect whether the user has set these options without having to dig through the log manually. Emit an INFO message since this is typically not critical.

### How Has This Been Tested?

Tested locally on Windows 11.

### Types of changes

- New feature (non-breaking change which adds functionality)

### Checklist:

- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

### CODE REVIEW

File: checks/network.py [hunk 1]
```diff
@@ -75,6 +75,20 @@ def checkDynamicBitrate(lines):
     return None
 
 
+def checkNetworkOptimizations(lines):
+    networkOptimizationLines = search('New socket loop enabled by user', lines)
+    if (len(networkOptimizationLines) > 0):
+        return [LEVEL_INFO, "Network Optimizations", "Network Optimizations are enabled. This is disabled by default. It is only suggested to enable Network Optimizations if you are having issues streaming. The setting can be toggled in Settings -> Advanced -> Network."]
```
Comment (Fenrirthviti) [inline checks/network.py:81]: `This is` -> `This option is` 

Unrelated to this PR, but to log at least somewhere public, we should have some kind of KB article for these settings that we can add here as links for the "what issues does this help with" part of things.

File: checks/network.py [hunk 2]
```diff
@@ -75,6 +75,20 @@ def checkDynamicBitrate(lines):
...
+    return None
+
+
+def checkTCPPacing(lines):
+    TCPPacingLines = search('Low latency mode enabled by user', lines)
+    if (len(TCPPacingLines) > 0):
+        return [LEVEL_INFO, "TCP Pacing", "TCP Pacing (Low Latency mode) is enabled. This is disabled by default. It is only suggested to enable TCP Pacing if you are having issues streaming. The setting can be toggled in Settings -> Advanced -> Network."]
```
Comment (Fenrirthviti) [inline checks/network.py:88]: same as above

**Review by Fenrirthviti — CHANGES_REQUESTED** [review]
Other than the nits, looks good to me.

File: checks/network.py (hunk 1 above)
Comment (RytoEX) [reply checks/network.py]: Addressed.

File: checks/network.py (hunk 2 above)
Comment (RytoEX) [reply checks/network.py]: Addressed.

**Review by Fenrirthviti — APPROVED** [review]

**Pull request merged**